db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/FromBaseTable.java

```diff
@@ -2709,14 +2713,18 @@ protected boolean supersetOfUniqueIndex(boolean[] eqCols) throws StandardExcepti
         ConglomerateDescriptor[] cds=tableDescriptor.getConglomerateDescriptors();
 
 		/* Cycle through the ConglomerateDescriptors */
+        IndexDescriptor id;
         for(ConglomerateDescriptor cd : cds){
-            if(!cd.isIndex()){
-                continue;
-            }
-            IndexDescriptor id=cd.getIndexDescriptor();
-
-            if(!id.isUnique()){
-                continue;
+            if(!cd.isIndex()) {
+                if (!cd.isPrimaryKey())
+                    continue;
+                else
+                    id = cd.getIndexDescriptor().getIndexDescriptor();
```

Comment:
`id = cd.getIndexDescriptor();` instead of `cd.getIndexDescriptor().getIndexDescriptor()`

Both achieve the same goal, so the smaller version is better imo.

Reply:
Thanks @arnaud-splice, I've updated the code accordingly.